test/tests/auxkernels/vector_postprocessor_visualization/vector_postprocessor_visualization.i

```diff
@@ -0,0 +1,82 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 10
+  ny = 10
+  parallel_type = REPLICATED
+  partitioner = linear
+[]
+
+[Variables]
+  [u]
+  []
+[]
+
+[Kernels]
+  [diff]
+    type = Diffusion
+    variable = u
+  []
+[]
+
+[BCs]
+  [left]
+    type = DirichletBC
+    variable = u
+    boundary = 'left'
+    value = 0
+  []
+  [right]
+    type = DirichletBC
+    variable = u
+    boundary = 'right'
+    value = 1
+  []
+[]
+
+[Executioner]
+  type = Steady
+  solve_type = PJFNK
+  petsc_options_iname = '-pc_type -pc_hypre_type'
+  petsc_options_value = 'hypre boomeramg'
+[]
+
+[Outputs]
+  exodus = true
+[]
+
+[AuxVariables]
+  [num_elems]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+  [partition_surface_area]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+[]
+
+[AuxKernels]
+  [wb_num_elems]
+    type = VectorPostprocessorVisualizationAux
+    vpp = 'wb'
+    vector_name = num_elems
+    variable = num_elems
+    execute_on = 'TIMESTEP_END'
+  []
+  [wb_partition_surface_area]
+    type = VectorPostprocessorVisualizationAux
+    vpp = 'wb'
+    vector_name = partition_surface_area
+    variable = partition_surface_area
+    execute_on = 'TIMESTEP_END'
+  []
+[]
+
+[VectorPostprocessors]
+  [wb]
+    type = WorkBalance
+    sync_to_all_procs = 'true'
+    execute_on = 'INITIAL'
```

Comment:
Shouldn't this be 'TIMESTEP_END' too if you are visualizing every timestep (see above)?

Reply:
Nope - the issue is a dependency resolution issue.

Reply:
Oh! Then this is a bug! VectorPostprocessors **should** be using the same PRE/POST logic as PostProcessors (and UserObjects). I'll checkout your branch and see if I can fix this.

Reply:
OK - So there isn't a bug here. Turns out that the check in `VectorPostprocessorVisualizationAux::timestepSetup()` is overzealous in that it runs in only one spot (e.g. at the beginning of the timestep). The only way to get around that is to run the VPP that you are trying to visualize _before_ that, which only leaves `INITIAL`. However, that defeats the purpose of the check since VPPs are allowed to change the size of their vectors with each invocation.